src/huggingface_hub/_commit_api.py

```diff
@@ -386,16 +391,57 @@ def _upload_lfs_object(operation: CommitOperationAdd, lfs_batch_action: dict, to
         return
     upload_action = lfs_batch_action["actions"].get("upload")
     verify_action = lfs_batch_action["actions"].get("verify")
-    with operation.as_file(with_tqdm=True) as fileobj:
-        logger.debug(f"Uploading {operation.path_in_repo} as LFS file...")
-        lfs_upload(
-            fileobj=fileobj,
-            upload_action=upload_action,
-            verify_action=verify_action,
-            upload_info=upload_info,
-            token=token,
+    use_hf_transfer = HF_HUB_ENABLE_HF_TRANSFER
+    if (
+        not (isinstance(operation.path_or_fileobj, str) or isinstance(operation.path_or_fileobj, Path))
+        and HF_HUB_ENABLE_HF_TRANSFER
+    ):
+        warnings.warn(
+            "hf_transfer is enabled but does not support uploading from bytes or BinaryIO, falling back to regular"
+            " upload"
         )
-        logger.debug(f"{operation.path_in_repo}: Upload successful")
+        use_hf_transfer &= False
```

Comment:
:llama_angry:

Yes it is effectively the same ahah.

Btw this makes me think about the fallback option, it will be very slow since we do not have the thread map in that case, wouldn't it be more user-friendly to throw an error ?

Reply:
*several files, don't have to be big to have a slow down

Personally I'd prefer an error than a "silent error" but it's up to you :)